testing/trino-product-tests-launcher/pom.xml

```diff
@@ -21,6 +21,13 @@
         <dependency>
             <groupId>io.trino</groupId>
             <artifactId>trino-plugin-reader</artifactId>
+            <exclusions>
+                <exclusion>
+                    <!-- trino main pulls a lot of dependencies pumping jar size, and we only need categories definitions defined in PluginReader class -->
+                    <groupId>io.trino</groupId>
+                    <artifactId>trino-main</artifactId>
```

Comment:
Also, by their design, prroduct tests should be "black box". They intentionally never depended on the trino-main.
Can you perhaps enhance the comment to indicate that too?

Also, why does the plugin reader pull trino-main?
Is it a fake dep? Does plugin reader combine different functionalities, some depending on trino-main and some not? maybe plugin reader should be two separate modules?

Reply:
PluginReader uses this single function from `trino-main` - `extractFunctions`: https://github.com/trinodb/trino/blob/master/testing/trino-plugin-reader/src/main/java/io/trino/server/PluginReader.java#L136

The product tests launcher only needs a few constants from the plugin reader. It definitely feels like something should be moved to improve dependencies, but I'm not sure what.

Reply:
> The product tests launcher only needs a few constants from the plugin reader. I

Can we move these constants out to `trino-plugin-reader-{api,spi,something}` jar?

Reply:
Yeah - it crossed my mind too. I was not convinced that we want extra module just for that. Probably the cleanest approach yet.
I will merge this one as a stop-gap though.